packages/react-instantsearch-core/src/core/indexUtils.js

```diff
@@ -69,24 +69,27 @@ export function refineValue(
 function refineMultiIndex(searchState, nextRefinement, context, resetPage) {
   const page = resetPage ? { page: 1 } : undefined;
   const indexId = getIndexId(context);
-  const state = has(searchState, `indices.${indexId}`)
-    ? {
-        ...searchState.indices,
-        [indexId]: {
-          ...searchState.indices[indexId],
-          ...nextRefinement,
-          ...page,
-        },
-      }
-    : {
-        ...searchState.indices,
-        ...{
+  const state =
+    searchState &&
+    searchState.indices &&
+    searchState.indices.hasOwnProperty(indexId)
```

Comment:
```suggestion
    searchState.indices[indexId]
```

The only value possible for `indices` is an object it should be safe to call it with the bracket notation.

Reply:
I agree, although it would mean the value can no longer be falsy